code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,197 @@
+/client/proc/cmd_select_equipment(mob/M in GLOB.mob_list)
+	set category = "Admin.Events"
+	set name = "Select equipment but better"
+
+
+	var/datum/select_equipment/ui  = new(usr, M)
+	ui.ui_interact(usr)
+
+/datum/select_equipment
+	var/client/user
+	var/mob/target
+
+	var/dummy_key
+	var/mob/living/carbon/human/dummy/dummy
+
+	var/static/list/cached_outfits
+
+	//normally a path; an initialized outfit object if it's a custom outfit
+	var/datum/outfit/selected_outfit = /datum/outfit
+	//used to keep track of which outfit the UI has selected
+	var/selected_name = "/datum/outfit"
+
+/datum/select_equipment/New(usr, mob/M)
+	user = CLIENT_FROM_VAR(usr)
+
+	if(!(ishuman(M) || isobserver(M)))
+		alert("Invalid mob")
+		return
+	target = M
+
+/datum/select_equipment/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SelectEquipment", "Select Equipment")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/select_equipment/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/select_equipment/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target))
+		return UI_CLOSE
+	return ..()
+
+/datum/select_equipment/ui_close(mob/user)
+	clear_human_dummy(dummy_key)
+	qdel(src)
+
+/datum/select_equipment/proc/init_dummy()
+	dummy_key = "selectequipmentUI_[target]"
+	dummy = generate_or_wait_for_human_dummy(dummy_key)
+	var/mob/living/carbon/C = target
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names